website/docs/r/key_vault_access_policy.html.markdown

```diff
@@ -71,6 +71,8 @@ The following arguments are supported:
 
 * `object_id` - (Required) The object ID of a user, service principal or security group in the Azure Active Directory tenant for the vault. The object ID of a service principal can be fetched from  `azuread_service_principal.object_id`. The object ID must be unique for the list of access policies. Changing this forces a new resource to be created.
 
+~> **NOTE:** Due to design limitation, setting `object_id` to `identity[0].principal_id` of another resource could not be done at the same time while updating the identity of the referenced resource to System Assigned Identity.
```

Comment:
This isn't a design limitation, but a behavioural bug in Terraform Core (since Terraform core isn't marking all computed blocks as SetNew) that needs to be worked around internally within the provider - so this isn't something we should be documenting in random places across the provider since unfortunately this can be present in any computed block.

FWIW I've got a task on my list to provide a fix for this, it's unfortunately blocked behind some other items - whilst the fix is relatively simple (logically) the concern is the knock-on effects this'll have across the provider, which is why we want to spend a little more time digging into this one.

Unfortunately in the interim we're kind of between a rock and a hard place, we can't realistically introduce notes for this through the documentation since it applies to any computed field - so we'd end up putting this all over the place. On the other hand the fix outlined above is logically simple to fix, however has broader implications across the provider which need to be investigated - so we'll be solving this for all computed fields at once, rather than just the `identity` fields, but we need to validate the impact of that change.

For the moment since we're intentionally not adding call-out's to the documentation as described above, whilst I'd like to thank you for this contribution - I'm going to close this PR for the moment, but we'll be looking to consolidate the issues outlined in #21545 (and the other issues) in the near future so that we can provide a fix for this.

Thanks!

Reply:
Thanks @tombuildsstuff for the explanations and good to hear the fix is being worked on